angularSTEP/src/app/setup/setup.component.ts

```diff
@@ -106,4 +107,21 @@ export class SetupComponent implements OnInit {
         });
     }
   }
+
+  formatUsername(username: string) {
+    const formattedUsername = username.replace(" ", "");
```

Comment:
Perhaps replacing with a - or something else would be preferable? What about other characters that aren't spaces and shouldn't be in the username?